### ISSUE TITLE
[qt5-static] Static Qt5 still pulls in some dynamic MinGW-specific libs

### ISSUE THREAD

**Issue by Sonic-Amiga** [msys2/MINGW-packages#19891]
### Description / Steps to reproduce the issue

As a workaround to https://github.com/msys2/MINGW-packages/issues/19890 i decided to try building for qt5-static. That one does have QSerialPort, but suddenly my app isn't entirely statically linked. It still refers to two mingw-specific DLLs:
- libzstd.dll
- zlib1.dll
The resulting .exe doesn't work without them.

### Expected behavior

No DLL dependencies, except Windows core DLLs

### Actual behavior

Extra MinGW-specific DLLs required

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-19045

### MINGW environments affected

- [X] MINGW64
- [ ] MINGW32
- [ ] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

### Are you willing to submit a PR?

Sorry, i am no expert here :(

**Comment by MehdiChinoune** [comment]
Could you show minimal reproducer.

**Comment by Sonic-Amiga** [comment]
I will, need some time for this. Will come back when done.

**Comment by Biswa96** [comment]
There were some similar issues about this topic.

**Comment by MehdiChinoune** [comment]
> There were some similar issues about this topic.

They were about qt6-static

**Comment by Sonic-Amiga** [comment]
Does it change the synopsis ? I guess no :)

**Comment by MehdiChinoune** [comment]
> Does it change the synopsis ? I guess no :)

What do you mean?

**Comment by Sonic-Amiga** [comment]
I mean the issue is still there; and it's good that it's also confirmed for Qt6. The easier it is.
I'll provide a sample later this week, sorry, busy with the main project.

**Comment by Sonic-Amiga** [comment]
Hello again ! Sorry, was overwhelmed. Back to the topic.
I did some more research. The behavior originates from cmake. It checks for qt5-static and successfully finds it. In its turn, qt5-static depends on libz and libzstd. cmake pulls in .dll.a versions. If i manually edit resulting ninja.build ; and replace .dll.a suffixes with just .a, everything is fine and the app links completely statically.

Is there a way to tell cmake that i want static linkage ? I tried googling, could find no answer.
